includes/blocks/course-theme/class-lesson-video.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * File containing the Lesson_Video class.
+ *
+ * @package sensei
+ * @since
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+use \Sensei_Course;
+use \Sensei_Utils;
+use \Sensei_Frontend;
+
+/**
+ * Class Lesson_Video is responsible for rendering the Lesson video template block.
+ */
+class Lesson_Video {
+	/**
+	 * Block JSON file.
+	 */
+	const BLOCK_JSON_FILE = '/lesson-blocks/course-theme-lesson-video.block.json';
+
+	/**
+	 * Lesson_Actions constructor.
+	 */
+	public function __construct() {
+		add_filter( 'the_content', [ $this, 'remove_featured_video_content' ], 1 );
+		$block_json_path = Sensei()->assets->src_path( 'course-theme/blocks' ) . self::BLOCK_JSON_FILE;
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-lesson-video',
+			[
+				'render_callback' => [ $this, 'render' ],
+				'style'           => 'sensei-theme-blocks',
+			],
+			$block_json_path
+		);
+	}
+
+	/**
+	 * Renders the block.
+	 *
+	 * @param array $attributes The block attributes.
+	 *
+	 * @access private
+	 *
+	 * @return string The block HTML.
+	 */
+	public function render( array $attributes = [] ) : string {
+		remove_action( 'sensei_lesson_video', [ Sensei_Frontend::class, 'sensei_lesson_video' ] );
+		$lesson_id = Sensei_Utils::get_current_lesson();
+		$user_id   = get_current_user_id();
+
+		if ( empty( $lesson_id ) || empty( $user_id ) ) {
+			return '';
+		}
+
+		$course_id = Sensei()->lesson->get_course_id( $lesson_id );
+
+		if (
+			! Sensei_Course::is_user_enrolled( $course_id )
+			|| Sensei_Utils::user_completed_lesson( $lesson_id )
+		) {
+			return '';
+		}
+
+		$content = Sensei_Utils::get_featured_video_html( $lesson_id ) ?? '';
```

Comment:
Why don't we use the `$content` that is provided as the second argument in this function? I looked into the `get_featured_video_html` method and it doesn't seem like doing something special. Just rendering the block that is already been done and the result is a second argument to this method.

  Also, if `get_featured_video_html` is still necessary, then I think it's better for us if that method lived here, in `Lesson_Video` class.

Reply:
Okay that makes sense, but note that the second argument for `$content` will always be empty since the JS block has no save method.
I agree to your point about the `get_featured_video_html`

Reply:
I yield! I guys are probably right.